src/services/rsvp/rsvp-router.test.ts

```diff
@@ -0,0 +1,114 @@
+import { describe, expect, it, beforeEach } from "@jest/globals";
+import { TESTER, getAsAdmin, getAsAttendee, getAsStaff, putAsApplicant } from "../../testTools.js";
+import { DecisionStatus, DecisionResponse } from "../../database/decision-db.js";
+import Models from "../../database/models.js";
+
+// Before each test it'll add this to the database
+beforeEach(async () => {
+    Models.initialize();
+    await Models.DecisionInfo.create({
+        userId: TESTER.id,
+        status: DecisionStatus.ACCEPTED,
+        response: DecisionResponse.PENDING,
+    });
+});
+
+describe("GET /rsvp", () => {
+    //i need to test
+    // get /rsvp, gets own rsvp data
+    // get /rsvp/:user, test w/ without perms, nonexistent user
+    // put /rsvp w/ json req
+
+    //good
+    it("gives a UserNotFound error for an non-existent user", async () => {
+        await Models.DecisionInfo.deleteOne({
+            userId: TESTER.id,
+        });
+
+        const response = await getAsAttendee("/rsvp/").expect(400);
+
+        expect(JSON.parse(response.text)).toHaveProperty("error", "UserNotFound");
+    });
+
+    it("works for an attendee user", async () => {
+        const response = await getAsAttendee("/rsvp/").expect(200);
+
+        expect(JSON.parse(response.text)).toHaveProperty("userId", TESTER.id);
+    });
+
+    it("works for an staff user", async () => {
+        const response = await getAsStaff("/rsvp/").expect(200);
+
+        expect(JSON.parse(response.text)).toHaveProperty("userId", TESTER.id);
+    });
+
+    it("works for an admin user", async () => {
+        const response = await getAsAdmin("/rsvp/").expect(200);
+
+        expect(JSON.parse(response.text)).toHaveProperty("userId", TESTER.id);
+    });
+});
+
+describe("GET /rsvp/:USERID", () => {
+    it("redirects to / if caller doesn't have elevated perms", async () => {
+        const response = await getAsAttendee("/rsvp/" + +TESTER.id).expect(302);
+
+        expect(response.text).toBe("Found. Redirecting to /");
+    });
+
+    it("gets if caller has elevated perms (Staff)", async () => {
+        const response = await getAsStaff("/rsvp/" + TESTER.id).expect(200);
+
+        expect(JSON.parse(response.text)).toHaveProperty("userId", TESTER.id);
+        expect(JSON.parse(response.text)).toHaveProperty("status");
+        expect(JSON.parse(response.text)).toHaveProperty("response");
```

Comment:
You should only use `toHaveProperty` when you want to validate that a certain property exists but don't care about the rest of the object, like for errors.

Instead, you should use something like `expect(...).toMatchObject(someObjectDefinedGlobally)` since you have multiple parameters here. This will create cleaner diffs when something goes wrong. If you don't want explicit matching (allowing other properties to exist), you can use `.toMatchObject(expect.objectContaining(someObjectDefinedGlobally))` instead.

Not going to annotate all instances of this, but you should fix this.